### PR TITLE
Document use of smaller storage variables

### LINKED ISSUE #509: uint8 gas cost higher than uint gas cost documentation
I don't believe we have this in the docs yet, but we should put this in the "gotchya" thread. I thought Peter's explanation was spot on. 

http://ethereum.stackexchange.com/questions/3067/why-uint8-takes-more-gas-than-uint256/3111#3111

### PR DESCRIPTION
**Pull request by Denton-L**
Close #509 

cc @chriseth

### CODE REVIEW

File: docs/miscellaneous.rst [hunk 1]
```diff
@@ -15,6 +15,23 @@ Statically-sized variables (everything except mapping and dynamically-sized arra
 - If an elementary type does not fit the remaining part of a storage slot, it is moved to the next storage slot.
 - Structs and array data always start a new slot and occupy whole slots (but items inside a struct or array are packed tightly according to these rules).
 
+.. warning::
+    When using elements that are smaller than 32 bytes, your contract's gas usage may be higher.
+    This is because the EVM operates on 32 bytes a a time. Therefore, if the element is smaller than
```
Comment (chriseth) [inline docs/miscellaneous.rst]: Typo: `at a time`

File: docs/miscellaneous.rst [hunk 2]
```diff
@@ -15,6 +15,23 @@ Statically-sized variables (everything except mapping and dynamically-sized arra
...
+    that, the EVM must use more operations in order to reduce the size of the element from 32 bytes
+    to the desired size.
+
+    It is only beneficial to use reduced-size arguments if you are dealing with storage values
+    because the compiler will pack multiple elements into one storage slot. When dealing with
```
Comment (chriseth) [inline docs/miscellaneous.rst]: `... and thus combine multiple reads or writes into a single operation`

**Comment by Denton-L** [comment]
Alright, added that stuff in!

File: docs/miscellaneous.rst [hunk 3]
```diff
@@ -15,6 +15,23 @@ Statically-sized variables (everything except mapping and dynamically-sized arra
...
+    function arguments or memory values, there is no inherent benefit because the compiler does not
+    pack these values.
+
+    Finally, in order to allow the EVM to optimize for this, ensure that you try to order your
+    storage variables such that they can be packed tightly. For example, declaring your storage
```
Comment (chriseth) [inline docs/miscellaneous.rst]: `storage variables (or struct members)`

File: docs/miscellaneous.rst [hunk 4]
```diff
@@ -15,6 +15,23 @@ Statically-sized variables (everything except mapping and dynamically-sized arra
...
+    than that, the EVM must use more operations in order to reduce the size of the element from 32
+    bytes to the desired size.
+
+    It is only beneficial to use reduced-size arguments if you are dealing with storage values
+    because the compiler will pack multiple elements into one storage slot, and thus, combine
+    multiple reads or writes into a single operation. When dealing with function arguments or memory
+    values, there is no inherent benefit because the compiler does not pack these values.
+
+    Finally, in order to allow the EVM to optimize for this, ensure that you try to order your
+    storage variables such that they can be packed tightly. For example, declaring your storage
```
Comment (chriseth) [inline docs/miscellaneous.rst]: `storage variables (or struct members)`

**Pull request merged**